x-pack/plugin/ilm/qa/multi-node/src/javaRestTest/java/org/elasticsearch/xpack/TimeSeriesRestDriver.java

```diff
@@ -396,6 +398,18 @@ public static String getSnapshotState(RestClient client, String snapshot) throws
         return (String) snapResponse.get("state");
     }
 
+    @Nullable
+    public static String waitAndGetShrinkIndexNameWithExtraClusterStateChange(RestClient client, String originalIndex)
```

Comment:
Can you please document `the why` behind this? (the cluster state batching and such)

Reply:
Ah very good point! On it.